server/src/main/java/org/apache/druid/client/BrokerServerView.java

```diff
@@ -106,17 +106,34 @@ public BrokerServerView(
     this.baseView = baseView;
     this.tierSelectorStrategy = tierSelectorStrategy;
     this.emitter = emitter;
-    this.segmentWatcherConfig = segmentWatcherConfig;
     this.clients = new ConcurrentHashMap<>();
     this.selectors = new HashMap<>();
     this.timelines = new HashMap<>();
 
+    // Validate and set the segment watcher config
+    this.segmentWatcherConfig = segmentWatcherConfig;
+    if (segmentWatcherConfig.getWatchedTiers() != null
```

Comment:
i know this isn't new, but I wonder since we now doing some validation here if we should also explode if `watchedTiers` is empty, which also seems like an invalid configuration

Reply:
Do you think it would make sense to add a `validate` method in `BrokerSegmentWatcherConfig` and do all these validations there or should we keep them here itself?